k8s/main.go

```diff
@@ -104,6 +127,59 @@ func main() {
 		os.Exit(1)
 	}
 
+	if metricsConfigFile != "" {
+		obs, err := initMetrics(metricsConfigFile)
+		if err != nil {
+			setupLog.Error(err, "unable to initialize metrics")
+			os.Exit(1)
+		}
+		defer shutdownMetrics(obs)
+	}
+
+	apiClientOptions := []utils.ApiClientOption{
+		utils.WithCertAuth(apiCertPath),
+	}
+
+	if utils.ShouldUseSATokens() {
+		apiClientOptions = append(apiClientOptions, utils.WithServiceAccountToken())
+	} else {
+		apiClientOptions = append(apiClientOptions, utils.WithUserPassword(ctx, "admin", ""))
+	}
+
+	apiClient, err := utils.NewAPIClient(
+		ctx,
+		utils.GetSymphonyAPIAddressBase(),
+		apiClientOptions...,
+	)
+	if err != nil {
+		setupLog.Error(err, "unable to create api client")
+		os.Exit(1)
+	}
+
+	pollInterval, err := time.ParseDuration(pollIntervalString)
+	if err != nil {
+		setupLog.Error(err, "unable to parse poll interval")
+		os.Exit(1)
+	}
+
+	reconcileInterval, err := time.ParseDuration(reconcileIntervalString)
+	if err != nil {
+		setupLog.Error(err, "unable to parse reconcile interval")
+		os.Exit(1)
+	}
+
+	deleteTimeOut, err := time.ParseDuration(deleteTimeOutString)
+	if err != nil {
+		setupLog.Error(err, "unable to parse delete timeout")
+		os.Exit(1)
+	}
+
+	deleteSyncDelay, err := time.ParseDuration(deleteSyncDelayString)
```

Comment:
What's deleteSyncDelay?

Reply:
This is used to let ARM to sync the failure status back. If the delete fails, before the finalizer is removed, it will wait for N sec to allow arm to sync the status before finally being removed from the cluster.

Because this delay is only needed in ARM, so we set the default value as 0. And we will expose it as a helm chart values and arc extension can configure a longer delay to let resource sync operator to sync the operation status.